app/models/episode_import.rb

```diff
@@ -183,4 +185,9 @@ def connection(u = uri)
   def account
     podcast_import.try(:account)
   end
+
+  def lookup_and_assign_episode!
+    lookup = Episode.find_by_item_guid(guid)
```

Comment:
Oh... erm... this should be scoped to this single Podcast.  Episode `item_guid` is only unique _within a single Podcast_.  You could have some other podcast with the same episode item_guid.